ksql-engine/src/main/java/io/confluent/ksql/planner/plan/FilterNode.java

```diff
@@ -31,16 +32,20 @@
 
   private final PlanNode source;
   private final Expression predicate;
+  private final List<SelectExpression> selectExpressions;
 
   public FilterNode(
       final PlanNodeId id,
       final PlanNode source,
-      final Expression predicate
+      final Expression predicate,
+      final List<SelectExpression> selectExpressions
```

Comment:
Is there a need to pass the select expressions separately? Why not just get them from `planNode.getSelectExpressions()`.?

Reply:
done